### PR TITLE
:bug: Fix open-quick-settings button position

### PR DESCRIPTION
**Pull request by fundon**

### CODE REVIEW

**Comment by fundon** [comment]
@abe33 
quick-settings

**Comment by abe33** [comment]
I still can't reproduce it, you should double check your stylesheet because there should be no need for that. 

screen

But the change isn't that big, so I guess it's okay. If you could just change the tests description so that it matches the convention used in the rest of the suite (if you need help on that please tell me, I'll take over your branch to fix this)

**Comment by fundon** [comment]
It's happend when `soft-wrap` was disabled,

**Comment by abe33** [comment]
Ok, in that case your tests are in the wrong place. It should not appear in the `when minimap.adjustMinimapWidthToSoftWrap is true` describe block but rather in the `when minimap.displayPluginsControls setting is true` (the quick settings section in the tests).

**Comment by fundon** [comment]
I moved it to the  `quick settings` section.  

I'm looking at the [quick_settings_custom_element](https://github.com/fundon/atom-minimap/tree/feature/quick_settings_custom_element) branch. Nice feature! :+1:

**Comment by abe33** [comment]
I'm not sure what's going on, but I can't reproduce this issue at all in my branch. For what I see by running your tests, the issue is that the controls is still doing the same width than the minimap, and it doesn't have any styles on it so I doubt that the test case is working properly (there should be an update that sets the width and margin of the controls).

If you want to compare with my current version, the equivalent of your test should be this one: 

https://github.com/fundon/atom-minimap/blob/feature/quick_settings_custom_element/spec/minimap-element-spec.coffee#L727-L731

Put a focus on it and comment the `querySelector` call like described here: https://github.com/fundon/atom-minimap/blob/feature/quick_settings_custom_element/spec/minimap-element-spec.coffee#L75-L77

You should see this result: 

screen

This is how it should behave, the margin trick used on the minimap is only necessary because flex box isn't reliable and setting a width on the minimap doesn't resize it properly. But nothing other than that should have an offset.

**Comment by fundon** [comment]
Ok, I think I should focus on your branch.
The `quick-settings` is being refactored and improved, so I should be doing something more important.

**Comment by fundon** [comment]
Hmm, I reopen it.
Because in `feature/quick_settings_custom_element`  branch has same issue.
Sorry, I didn't explain clearly before.

The `minimap-scroll-indicator` and  `open-minimap-quick-settings`'s position was broken in this screenshot.

File: lib/minimap-element.coffee [hunk 1]
```diff
@@ -267,6 +267,7 @@ class MinimapElement extends HTMLElement
 
     @applyStyles @controls,
       width: Math.min(@canvas.width, @width) + 'px'
+      'margin-left': (@marginRight || 0) + 'px'
```
Comment (fundon) [inline lib/minimap-element.coffee]: Or use this:

```coffee
    @applyStyles @controls,
      width: Math.min(@canvas.width, @width) + (0 + @marginRight) + 'px'
```

Comment (abe33) [inline lib/minimap-element.coffee]: More than a fix, I really need a reproducible test case for the test suite. As I already explained, the solution IMO is not to add margin to the controls div. That div should always have the size of the smallest element in the minimap (either the `atom-text-editor-minimap` or `canvas` element), this is the line 269 above. If this doesn't work in some situation, the solution should be to fix this, not to add another offset somewhere.

In your screen above, does the controls div have a style attribute?

Also, do you see in the following test anything that doesn't match the configuration in which it should fail?

https://github.com/fundon/atom-minimap/blob/feature/quick_settings_custom_element/spec/minimap-element-spec.coffee#L702-L731

**Comment by fundon** [comment]
That's my test output below on my machine. 
```
[67623:0214/231648:WARNING:dns_config_service_posix.cc(151)] dns_config has unhandled options!
[67625:0214/231649:INFO:renderer_main.cc(211)] Renderer process started
[67623:0214/231653:INFO:CONSOLE(79)] "Window load time: 3730ms", source: file:///Applications/Atom.app/Contents/Resources/app/static/index.js (79)
..............................................Error Downloading Update: The request timed out.
.....FFF................................................................................

MinimapElement
  when attached to the text editor element
    when minimap.displayPluginsControls setting is true
      when the adjustMinimapWidthToSoftWrap setting is enabled
        the open-minimap-quick-settings position
          positions the open-minimap-quick-settings button
            it right should be less than 2560
              Expected 2637.90625 to be less than 2560. (spec/minimap-element-spec.coffee:747:51)
              Expected 2637.90625 to equal 2556. (spec/minimap-element-spec.coffee:748:51)
          positions minimap-scroll-indicator
            it right should be <= 2560
              Expected 2641.90625 to be less than 2641. (spec/minimap-element-spec.coffee:768:49)
              Expected 2641.90625 to be less than 2560. (spec/minimap-element-spec.coffee:769:49)
          it positions the controls div over the canvas
            Expected 1020.734375 to equal 1016.734375. (spec/minimap-element-spec.coffee:731:40)


Finished in 128.999 seconds
134 tests, 177 assertions, 5 failures, 0 skipped

Tests failed
```

# travic-ci output
https://travis-ci.org/fundon/atom-minimap/builds/50755890

```

MinimapElement
  when attached to the text editor element
    when minimap.displayPluginsControls setting is true
      when the adjustMinimapWidthToSoftWrap setting is enabled
        the open-minimap-quick-settings position
          positions the open-minimap-quick-settings button
            it right should be less than 2560
              Expected 2556.90625 to equal 2556. (spec/minimap-element-spec.coffee:748:51)
          positions minimap-scroll-indicator
            it right should be <= 2560
              Expected 2560.90625 to be less than 2560. (spec/minimap-element-spec.coffee:769:49)
```

**Comment by fundon** [comment]
@abe33 
I guess the `devicepixelratio` is 1 in the Travis-ci.
https://discuss.atom.io/t/what-is-devicepixelratio-number-in-the-travis-ci/14901/1

**Comment by fundon** [comment]
Focus on `feature/quick_settings_custom_element` branch.

**Pull request closed without merging**